pkg/controller/metadata/metadata_controller.go

```diff
@@ -481,7 +236,7 @@ func (r *ReconcileMetadata) Reconcile(request reconcile.Request) (reconcile.Resu
 		logInfo("Created metadata secret", metadataSecret.ObjectMeta, "version", currentVersion)
 	} else if err != nil {
 		return reconcile.Result{}, err
-	} else if ShouldRegenerate(foundSecret, currentVersion, *instance) {
+	} else if shouldRegenerateMetadata(foundSecret, currentVersion, *instance, *currentSigningCertConfigMap) {
```

Comment:
I may well have misunderstood, but I wonder if carrying on regenerating the metadata _with the same secret_ is a bit wrong. I guess it's no more wrong that it was before, but now we're taking a more active decision to _be_ wrong?

Reply:
How do you mean? The metadata secret gets updated if it needs to be. It's just a container that has the details of the current metadata.

Reply:
Hmm I think I am confused about this. I thought there was a key in there somewhere (sounds like that's not the secret), but we're still re-using that same key pair even through regenerating the metadata and slapping a new date on it? Or not, mostly this is probably my misunderstanding a conversation the other day :-)

Reply:
Ah, I think you're thinking of how we're currently not rotating the private keys were using to sign the metadata of self-signed certs. We would like to do this but it would break stuff at the moment as there's no agreement in eIDAS about how to dual run stuff. It's in the list of things to do at some point, but not until eIDAS gives us a clearer steer on what that process should like like. For now, the first cert we're going to have to renew isn't until April.